pyflakes/checker.py

```diff
@@ -481,7 +481,8 @@ class Checker(object):
     del _customBuiltIns
 
     def __init__(self, tree, filename='(none)', builtins=None,
-                 withDoctest='PYFLAKES_DOCTEST' in os.environ):
+                 withDoctest='PYFLAKES_DOCTEST' in os.environ,
+                 scope=ModuleScope):
```

Comment:
do we even need this?

We could derive this from the `tree` 's `ast.Node` type.

Is there any scenario when the scope type shouldnt be derived from the `tree` type?
Less options is less to support.

Reply:
Many more scopes could be added to pyflakes. For eg: a `TryScope`. Thus deriving from tree type might hard code this.

Reply:
We worry about that then.